src/Poll.tsx

```diff
@@ -172,11 +173,8 @@ class ContextlessPoll<TData, TError> extends React.Component<
 
   private keepPolling = !this.props.lazy;
 
-  /**
-   * Abort controller to cancel the current fetch query
-   */
-  private abortController = new AbortController();
-  private signal = this.abortController.signal;
+  // List of requests to be cancelled when component unmounts
+  private pendingRequests: Array<CancelablePromise<any>> = [];
```

Comment:
Are you sure that is works as before? When I've checked the doc about how cancel a request, I found this `AbortController`, maybe cancelling the promise also works but I'm curious.

Also, I will prefer to have only one fetch running to avoid concurrencies issues (We don't want to have the result of the previous query if the second query is quicker). So ideally, if a previous request is running during a new fetch, you cancel the previous one :wink: This should also solve #12

Reply:
Yeah! this works fine.

I tried this approach initially because I needed a way to check whether the request is still going before changing the state. Later I saw that you can also check that information from the signal, that's why I closed the PR.